lib/index.js

```diff
@@ -59,52 +59,73 @@ shp.combine = function(arr) {
   return out;
 };
 shp.parseZip = function(buffer, whiteList) {
-  var key;
+  var key, i;
   buffer = toBuffer(buffer);
-  var zip = unzip(buffer);
-  var names = [];
-  whiteList = whiteList || [];
-  for (key in zip) {
-    if (key.indexOf('__MACOSX') !== -1) {
-      continue;
+  return unzip(buffer).then(function(zip) {
+    var names = [];
+    whiteList = whiteList || [];
+    if (Object.keys) {
```

Comment:
if you want to get rid of the for in loop, we can talk about that, but lets put that into a different pull